src/presentation/pdfFileEditor.tsx

```diff
@@ -191,22 +228,64 @@ export default class PdfFileEditor extends Component<PdfFileEditorProps, PdfFile
         const rectangle = this.getCropRectangle();
         if (rectangle) {
             const {left, top, right, bottom} = rectangle;
-            const x = startPos.x < (left + right) / 2 ? right : left;
-            const y = startPos.y < (top + bottom) / 2 ? bottom : top;
-            this.setState({adjustingCropRectangle: true, cropRectangle: [{x, y}, startPos]});
+            const {x: startX, y: startY} = startPos;
+            const margin = CROP_ADJUSTMENT_DRAG_MARGIN;
+            if (isPointWithinBounds(startX, startY, left - margin, top - margin, right + margin, bottom + margin)
+                && !isPointWithinBounds(startX, startY, left + margin, top + margin, right - margin, bottom - margin)) {
+                // Resize time!
+                const centerX = (left + right) / 2;
+                const centerY = (top + bottom) / 2;
+                let x = startPos.x < centerX ? right : left;
+                let y = startPos.y < centerY ? bottom : top;
+                this.setState({adjustingCropRectangle: CropAdjustment.RESIZING, cropRectangle: [{x, y}, startPos]});   
+            } else {
+                // Reposition time!
+                let a = { x: left, y: top };
+                let b = { x: right, y: bottom };
+                let dragStart = startPos;
+                this.setState({adjustingCropRectangle: CropAdjustment.POSITIONING, cropRectangle: [a, b, dragStart]});
+            }
         } else {
-            this.setState({adjustingCropRectangle: true, cropRectangle: [startPos, startPos]});
+            this.setState({adjustingCropRectangle: CropAdjustment.RESIZING, cropRectangle: [startPos, startPos]});
         }
     }
 
     onPan(_delta: ObjectVector2, position: ObjectVector2) {
-        if (this.state.cropRectangle) {
-            this.setState({cropRectangle: [this.state.cropRectangle[0], position]});
+        const { adjustingCropRectangle, cropRectangle } = this.state;
+        if (!isCropAdjusting(adjustingCropRectangle)) {
+            return;
+        } else if (!cropRectangle) {
+            // Invalid state! This should never happen :(
+            throw new Error('Crop is being adjusted but crop rectangle does not exist.');
+        }
+        // We are adjusting crop rectangles.
+        switch(adjustingCropRectangle) {
+            case CropAdjustment.POSITIONING: {
```

Comment:
I presume these braces are here to confine the scope of the variables in the case?  I hadn't seen that before, but it makes sense...